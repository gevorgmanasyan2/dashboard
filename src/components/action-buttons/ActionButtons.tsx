import { Button } from 'antd'
import React, { FC, ReactElement } from 'react'

interface ActionButtonsProps {
  buttonText: string
  className: string
  buttonType?: boolean
  icon?: ReactElement | null
  onClick?: (value: any) => void
}

const ActionButtons: FC<ActionButtonsProps> = ({ buttonType, buttonText, className, onClick,icon }) => {
  return (
    <Button className={className} disabled={buttonType} onClick={onClick} icon={icon}>
      {buttonText}
    </Button>
  )
}

export default ActionButtons
