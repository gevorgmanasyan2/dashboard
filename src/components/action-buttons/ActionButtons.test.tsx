/* eslint-disable no-undef */
import React from 'react'
import { shallow } from 'enzyme'
import ActionButtons from './ActionButtons'

it('should render ActionButtons component', () => {
  const mockItem = {
    buttonType: true,
    buttonText: 'save',
    className: 'cancel',
  }
  expect(shallow(<ActionButtons {...mockItem} />)).toMatchSnapshot()
})
