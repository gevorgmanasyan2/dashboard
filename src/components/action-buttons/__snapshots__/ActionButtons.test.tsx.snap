// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ActionButtons component 1`] = `
<Button
  block={false}
  className="cancel"
  disabled={true}
  ghost={false}
  htmlType="button"
  loading={false}
>
  save
</Button>
`;
